arch/arm64/src/common/arm64_syscall.c

```diff
@@ -397,11 +324,59 @@ int arm64_syscall(uint64_t *regs)
         break;
 #endif
 
+      /* This is not an architecture-specific system call.  If NuttX is built
+       * as a standalone kernel with a system call interface, then all of the
+       * additional system calls must be handled as in the default case.
+       */
+
       default:
+        {
+#ifdef CONFIG_LIB_SYSCALL
 
-          DEBUGPANIC();
-          break;
+          /* Verify that the SYS call number is within range */
+
+          DEBUGASSERT(cmd >= CONFIG_SYS_RESERVED && cmd < SYS_maxsyscall);
+
+          /* Make sure that there is a no saved SYSCALL return address.  We
+           * cannot yet handle nested system calls.
+           */
+
+          regs[REG_X0] = dispatch_syscall(regs[REG_X0], regs[REG_X1],
```

Comment:
This is completely unnecessary, the registers x0-x7 are already correct when coming from userspace. There is NO REASON TO READ THEM FROM THE CONTEXT AREA. What I had done is much more optimized than what this does.

Reply:
Another thing this change does; the system call now runs with interrupts disabled, affecting interrupt service latency.

Reply:
@lipengfei28 please take a look.